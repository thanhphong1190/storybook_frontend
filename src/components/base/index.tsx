export { Button, ButtonLink } from './Button'
export { SVGIcon, IconEnums } from './icon/svgIcon'
export { Icon, Icons } from './Icon'
