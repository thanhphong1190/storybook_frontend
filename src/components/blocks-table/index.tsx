export { BlocksTable, BlocksTableProps } from './BlocksTable'
