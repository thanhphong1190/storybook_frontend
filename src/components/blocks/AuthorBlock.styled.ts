import styled from 'styled-components'
import tw from 'twin.macro'

export const AuthorSection = styled.div`
  ${tw`relative flex flex-col items-center w-full md:items-start md:flex-row`}
`

export const AuthorAvatar = styled.a<{ avatarImgURL?: string }>`
  ${tw`relative block rounded-full`}
  background: url(${props => props.avatarImgURL}) 0 0/cover no-repeat;
  width: 123px;
  height: 123px;
  z-index:-1;
`

export const AuthorAvatarFrame = styled.div<{ frameImgURL: string }>`
  background: url(${props => props.frameImgURL}) -1px -1px/cover no-repeat;
  width: 125px;
  height: 125px;
`

export const AuthorInformation = styled.div`
  ${tw`relative w-full font-display md:pl-8 md:max-w-screen-md`}
`

export const AuthorName = styled.h4`
  ${tw`text-xl text-orange font-bold uppercase`}
`

export const AuthorBios = styled.h6`
  ${tw`text-base text-blue__dark font-bold`}
`

export const AuthorIntro = styled.div<{collapsed?: boolean}>`
  ${tw`text-grey__medium mt-4`}

  & p {
    ${props => props.collapsed && tw`truncate`}
  }
`

export const ReadButton = styled.a`
  ${tw`text-grey__medium mt-2 text-sm cursor-pointer`}

  &::before {
    content: "\\[";
  }

  &::after {
    content: "\\]";
  }
`

