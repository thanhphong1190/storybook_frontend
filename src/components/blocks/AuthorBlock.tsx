import {
  AuthorSection,
  AuthorAvatarFrame,
  AuthorAvatar,
  AuthorInformation,
  AuthorName,
  AuthorBios,
  AuthorIntro,
  ReadButton
} from './AuthorBlock.styled'
import circleFrame from '../../static/imgs/circle.png'
import { useState } from 'react'

export interface AuthorBlockProps {
  avatarURL: string
  authorName: string
  authorBios: string
  authorIntro: string
}

export const AuthorBlock = (props: AuthorBlockProps) => {
  const { avatarURL, authorName, authorBios, authorIntro } = props
  const [introCollapsed, setIntroCollapsed] = useState(true)
  const [readBtnText, setReadBtnText] = useState('read more')
  const expandDesc = () => {
    setIntroCollapsed(!introCollapsed)

    if (!introCollapsed) {
      setReadBtnText('read more')
    } else {
      setReadBtnText('close')
    }
  }
  return (
    <AuthorSection>
      <AuthorAvatarFrame frameImgURL={circleFrame}>
        <AuthorAvatar avatarImgURL={avatarURL} />
      </AuthorAvatarFrame>
      <AuthorInformation>
        <AuthorName>{authorName}</AuthorName>
        <AuthorBios>{authorBios}</AuthorBios>
        <AuthorIntro collapsed={introCollapsed}>
          <p dangerouslySetInnerHTML={{ __html: authorIntro }} />
        </AuthorIntro>
        <ReadButton onClick={expandDesc}>{readBtnText}</ReadButton>
      </AuthorInformation>
    </AuthorSection>
  )
}
export default AuthorBlock
