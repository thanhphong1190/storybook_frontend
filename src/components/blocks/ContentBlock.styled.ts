import styled from 'styled-components'
import tw from 'twin.macro'

export const ContentBlockText = styled.p<{fontBold?: boolean}>`
  ${tw`mb-8 font-display`}
  ${props => (!!!props.color || props.color == 'orange') && tw`text-orange`}
  ${props => props.color == 'blue' && tw`text-blue`}
  ${props => props.color == 'white' && tw`text-white`}
  ${props => props.color == 'black' && tw`text-black`}
  ${props => props.fontBold && tw`font-bold`}
`
