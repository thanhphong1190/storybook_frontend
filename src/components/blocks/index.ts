export { ContentBlock, ContentBlocksProps } from './ContentBlock'
export { AuthorBlock, AuthorBlockProps } from './AuthorBlock'
export { VideoBlock, PlayButton, VideoBlocksProps, VideoPlayer, VideoPlayerProps } from './VideoBlock'