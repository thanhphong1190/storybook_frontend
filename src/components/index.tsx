export { ContentBlock } from './blocks/ContentBlock'
export { Header } from './header/Header'
export { Navigation } from './navigation/Navigation'
export { HeroImage } from './hero/HeroImage'
export { KeyMessage } from './key-message/KeyMessage'
export { Footer } from './footer/Footer'
export { default as Banner } from './banner/Banner'
export { default as OrangeBanner } from './orange-banner/OrangeBanner'
export { ReviewCard, CustomerReviews } from './customer-reviews'
export { StandardContent } from './standard-content'
export { FranchiseBenefitCard, FranchiseBenefits } from './franchise-benefits'
export { Hero } from './hero/Hero'
export { ReadyStartedBlocks } from './ready-started-blocks'
export { BlocksTable } from './blocks-table'
