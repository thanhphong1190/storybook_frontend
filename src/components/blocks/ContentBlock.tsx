import { Button } from '../base'
import {
  Card,
  CardHeader,
  CardBody,
  CardFooter
} from '../../styles/global.styles'
import { ContentBlockText } from './ContentBlock.styled'

export interface ContentBlocksProps {
  headerText: string
  contentPrimayText: string
  contentSecondaryText: string
  primaryBtnText: string
  secondaryBtnText: string
  onPrimaryBtnClick?: Function
  onSecondaryBtnClick?: Function
}

export const ContentBlock = (props: ContentBlocksProps) => {

  const { headerText,
    contentPrimayText,
    contentSecondaryText,
    primaryBtnText,
    secondaryBtnText,
    onPrimaryBtnClick,
    onSecondaryBtnClick} = props

  return (
    <Card>
      <CardHeader separator>{headerText}</CardHeader>
      <CardBody>
        <ContentBlockText
          fontBold
          color='orange'
          dangerouslySetInnerHTML={{ __html: contentPrimayText }}
        />
        <ContentBlockText
          color='black'
          dangerouslySetInnerHTML={{ __html: contentSecondaryText }}
        />
      </CardBody>
      <CardFooter>
        <Button
          color='blue'
          onClick={() =>
            onSecondaryBtnClick && onSecondaryBtnClick()
          }
        >
          {secondaryBtnText}
        </Button>
        <Button
          onClick={() => onPrimaryBtnClick && onPrimaryBtnClick()}
        >
          {primaryBtnText}
        </Button>
      </CardFooter>
    </Card>
  )
}

export default ContentBlock
