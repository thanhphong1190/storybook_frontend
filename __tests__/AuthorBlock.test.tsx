import { AuthorBlock, AuthorBlockProps } from '../src/components/blocks'
import renderer from 'react-test-renderer'
import fionaAvatarImg from '../static/example/Fiona.jpg'

const props: AuthorBlockProps = {
  avatarURL: fionaAvatarImg,
  authorName: 'Jasmine Green',
  authorBios:
    'Certificate 3 & 4 in fitness -Punch-fit boxing trainer -Bio-print practitioner',
  authorIntro: `Jasmine is a transformation specialist and has over 5 years experience in the fitness industry as a personal trainer.
      My first contact with Plus Fitness 24/7 was as a long time member of their health clubs, which have been around for many years and were well known in the Macarthur region. 
      I have spent my career in business banking and was fortunate enough to have Plus Fitness 24/7 as a client when they made the decision to develop the brand as a franchise. 
      It was very exciting seeing the process and watch the company sell it's first franchise,
      and before I knew it they had sold their 50th and so on. I had a relationship with some of the early franchisees and they had such great things to say about their experiences, 
      I developed a keen interest in the model and contemplated jumping ship many times from business banker to business owner.`
}

it('renders correctly', () => {
  const tree = renderer.create(<AuthorBlock {...props} />).toJSON()
  expect(tree).toMatchSnapshot()
})
