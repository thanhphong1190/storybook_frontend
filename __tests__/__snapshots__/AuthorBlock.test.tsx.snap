// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="AuthorBlockstyled__AuthorSection-nrpm2a-0 fFGFPd"
>
  <div
    className="AuthorBlockstyled__AuthorAvatarFrame-nrpm2a-2 klgvKL"
  >
    <a
      className="AuthorBlockstyled__AuthorAvatar-nrpm2a-1 kPZjML"
    />
  </div>
  <div
    className="AuthorBlockstyled__AuthorInformation-nrpm2a-3 djVNPu"
  >
    <h4
      className="AuthorBlockstyled__AuthorName-nrpm2a-4 hDoxlh"
    >
      Jasmine Green
    </h4>
    <h6
      className="AuthorBlockstyled__AuthorBios-nrpm2a-5 eTANDM"
    >
      Certificate 3 & 4 in fitness -Punch-fit boxing trainer -Bio-print practitioner
    </h6>
    <div
      className="AuthorBlockstyled__AuthorIntro-nrpm2a-6 cBGRtl"
    >
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "Jasmine is a transformation specialist and has over 5 years experience in the fitness industry as a personal trainer.
      My first contact with Plus Fitness 24/7 was as a long time member of their health clubs, which have been around for many years and were well known in the Macarthur region. 
      I have spent my career in business banking and was fortunate enough to have Plus Fitness 24/7 as a client when they made the decision to develop the brand as a franchise. 
      It was very exciting seeing the process and watch the company sell it's first franchise,
      and before I knew it they had sold their 50th and so on. I had a relationship with some of the early franchisees and they had such great things to say about their experiences, 
      I developed a keen interest in the model and contemplated jumping ship many times from business banker to business owner.",
          }
        }
      />
    </div>
    <a
      className="AuthorBlockstyled__ReadButton-nrpm2a-7 ioOqnY"
      onClick={[Function]}
    >
      read more
    </a>
  </div>
</div>
`;
